test/e2e/toolchaincluster_test.go

```diff
@@ -55,13 +55,16 @@ func verifyToolchainCluster(t *testing.T, await *wait.Awaitility, otherAwait *wa
 
 		// then the ToolchainCluster should be ready
 		require.NoError(t, err)
-		_, err = await.WaitForNamedToolchainClusterWithCondition(toolchainCluster.Name, wait.ReadyToolchainCluster)
+		namedToolchainCluster, err := await.WaitForNamedToolchainClusterWithCondition(toolchainCluster.Name, wait.ReadyToolchainCluster)
 		require.NoError(t, err)
+		checkAbsenceOfTenantRoleLabel(t, &namedToolchainCluster)
 		// other ToolchainCluster should be ready, too
-		_, err = await.WaitForToolchainClusterWithCondition(otherAwait.Type, otherAwait.Namespace, wait.ReadyToolchainCluster)
+		toolChainClusterWithCondition, err := await.WaitForToolchainClusterWithCondition(otherAwait.Type, otherAwait.Namespace, wait.ReadyToolchainCluster)
 		require.NoError(t, err)
-		_, err = otherAwait.WaitForToolchainClusterWithCondition(await.Type, await.Namespace, wait.ReadyToolchainCluster)
+		checkAbsenceOfTenantRoleLabel(t, &toolChainClusterWithCondition)
+		otherToolchainCluster, err := otherAwait.WaitForToolchainClusterWithCondition(await.Type, await.Namespace, wait.ReadyToolchainCluster)
 		require.NoError(t, err)
+		checkAbsenceOfTenantRoleLabel(t, &otherToolchainCluster)
 	})
```

Comment:
I agree with @xcoulon that calling the `checkAbsenceOfTenantRoleLabel` sounds a bit weird in this test. 
These two tests create a kind of dummy ToolchainClusters (they don't use the script from the CICD repo), so what is not set as part of the test, then it won't be present in the actual CR. 
The main purpose of these tests is to verify that the controller and other ToolchainCluster CRs will work as expected based on what is set in the dummy (newly created) ToolchainCluster CR - they have the right conditions.
TBH, I would drop the `checkAbsenceOfTenantRoleLabel` from it as it doesn't have any impact on the result of the test. There is no logic that we would verify with it.

Reply:
change in [475eb56](https://github.com/codeready-toolchain/toolchain-e2e/pull/645/commits/475eb56f9ddcb81ffb5a99e6b9d3e40183d987df)